src/components/proxy/xdp/process.rs

```diff
@@ -0,0 +1,606 @@
+use crate::{
+    components::proxy::PipelineError,
+    filters::{self, Filter as _},
+    metrics,
+    net::EndpointAddress,
+};
+pub use quilkin_xdp::xdp;
+use quilkin_xdp::xdp::{
+    packet::{
+        csum,
+        net_types::{FullAddress, IpAddresses, NetworkU16, UdpPacket},
+        Packet, PacketError,
+    },
+    HeapSlab, Umem,
+};
+use std::{
+    net::{IpAddr, SocketAddr},
+    sync::{
+        atomic::{AtomicU16, Ordering},
+        Arc,
+    },
+};
+
+struct QPacket {
```

Comment:
Can this be renamed to be more clear? Like could it be called what layer of packet it is, e.g. IPPacket, EthernetPacket, etc. Can there be docs on what it's purpose are here?